include/RAJA/config.hpp.in

```diff
@@ -514,7 +524,7 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 
 // This is the same as undefined compiler, but squelches the warning message
 #elif defined(RAJA_COMPILER_MSVC)
-
+#define RAJA_MAX_ALIGN alignof(std::max_align_t)
```

Comment:
should this be `#define RAJA_MAX_ALIGN 16`?

Reply:
MSVC throws a hard compiler error if the alignment is set to anything larger than 8 bytes. Since the issue only manifests in Clang since alignof(std::max_align_t) being tied to size(long double) is specific to that compiler this seemed like a reasonable approach.

Reply:
What error are you getting? The only time an alignas directive should be ill formed is if the alignment specified is lower than the native alignment of the type or object, and MSVC doesn't produce an error for that.

Reply:
The old build logs are gone but it was some form of:

static_assert failed: 'You've instantiated std::aligned_storage<Len, Align> with an extended alignment (in other words, Align > alignof(max_align_t)).
...

Please define either (1) _ENABLE_EXTENDED_ALIGNED_STORAGE to acknowledge that you understand this message and that you actually want a type with an extended alignment, or (2) _DISABLE_EXTENDED_ALIGNED_STORAGE

Seemed like keeping the previous definition made sense for this case if the issue is contained to Clang. But could be persuaded otherwise.